server/notification-providers/heii-oncall.js

```diff
@@ -0,0 +1,67 @@
+const { UP, DOWN, getMonitorRelativeURL } = require("../../src/util");
+const { setting } = require("../util-server");
+
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+class HeiiOnCall extends NotificationProvider {
+    name = "HeiiOnCall";
+
+    /**
+     * @inheritdoc
+     */
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        const payload = heartbeatJSON || {};
+
+        const baseURL = await setting("primaryBaseURL");
+        if (baseURL && monitorJSON) {
+            payload["url"] = baseURL + getMonitorRelativeURL(monitorJSON.id);
+        }
+
+        try {
+            if (!heartbeatJSON) {
+                // Testing or general notification like certificate expiry
+                payload["msg"] = msg;
+                return this.postNotification(notification, "alert", payload);
+            }
+
+            if (heartbeatJSON.status === DOWN) {
+                return this.postNotification(notification, "alert", payload);
+            }
+
+            if (heartbeatJSON.status === UP) {
+                return this.postNotification(notification, "resolve", payload);
+            }
+        } catch (error) {
+            this.throwGeneralAxiosError(error);
+        }
+
+    }
+
+    /**
+     * Post to Heii On-Call
+     * @param {BeanModel} notification Message title
+     * @param {string} action Trigger action (alert, resovle)
+     * @param {object} payload Data for Heii On-Call
+     * @returns {Promise<string>} Success message
+     */
+    async postNotification(notification, action, payload) {
+        const config = {
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
+                Authorization: "Bearer " + notification.heiiOnCallApiKey,
+            },
+        };
+
+        // Post to Heii On-Call Trigger https://heiioncall.com/docs#manual-triggers
+        await axios.post(
+            `https://heiioncall.com/triggers/${notification.heiiOnCallTriggerId}/${action}`,
+            payload,
+            config
+        );
+
+        return "Sent Successfully";
+    }
```

Comment:
This should refactor the code to be closer to what the other notification providers look like => keeping it more consistent.

```suggestion
    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
        const okMsg = "Sent Successfully.";
        const payload = heartbeatJSON || {};

        const baseURL = await setting("primaryBaseURL");
        if (baseURL && monitorJSON) {
            payload["url"] = baseURL + getMonitorRelativeURL(monitorJSON.id);
        }

        const config = {
            headers: {
                Accept: "application/json",
                "Content-Type": "application/json",
                Authorization: "Bearer " + notification.heiiOnCallApiKey,
            },
        };
        
        const heiiUrl = `https://heiioncall.com/triggers/${notification.heiiOnCallTriggerId}/`;
        
        // docs https://heiioncall.com/docs#manual-triggers
        try {
            if (!heartbeatJSON) {
                // Testing or general notification like certificate expiry
                payload["msg"] = msg;
                await axios.post(heiiUrl + "alert", payload, config);
                return okMsg;
            }

            if (heartbeatJSON.status === DOWN) {
                await axios.post(heiiUrl + "alert", payload, config);
                return okMsg;
            }
            if (heartbeatJSON.status === UP) {
                await axios.post(heiiUrl + "resolve",payload, config);
                return okMsg;
            }
        } catch (error) {
            this.throwGeneralAxiosError(error);
        }
    }
```